centaur/src/main/scala/centaur/serialization/TestsReports.scala

```diff
@@ -0,0 +1,76 @@
+package centaur.serialization
+
+import java.io.{BufferedWriter, FileNotFoundException, FileWriter}
+
+import org.slf4j.LoggerFactory
+
+import scala.collection.concurrent.TrieMap
+import scala.io.Source
+import scala.util.{Failure, Success, Try}
+
+/**
+  * This class contains information about last version in which corresponding integration test was successfully run
+  *
+  * @param testCasesRuns key is a test name; value is a version of Cromwell, in which this test passed last time
+  */
+case class TestsReports(testCasesRuns: TrieMap[String, String]) {
+  def addSuccessfulTest(testName: String, version: String): Option[String] =
+    testCasesRuns.put(testName, version)
+
+  def testsToSkip(version: String): Set[String] =
+    testCasesRuns.filter(_._2 == version).keys.toSet
+}
+
+/**
+  * Serializer for {@link centaur.serialization.TestsReports} class. Based on Circe, serializes to JSON
+  */
+object TestsReportsSerializer {
+  private val logger = LoggerFactory.getLogger(TestsReportsSerializer.getClass)
+
+  def read(fileName: String): TestsReports = {
+    import io.circe.parser._
+
+    val obj = readFile(fileName) {
+      case Success(res) => decode[TrieMap[String, String]](res.getLines.mkString) match {
```

Comment:
Why are you reading and writing `TrieMap` instead of case class?

Reply:
Just in order to not implement my own Encoder/Decoder for `TestsReports` and use the standard one, because reading/writing and creation of `TestsReports` instances is encapsulated anyway, and this way is just less work to do.

Reply:
Thanks to Circe, all this stuff is done automatically and implicitly
https://circe.github.io/circe/codecs/auto-derivation.html

Reply:
Forgot to import `generic.auto`. Yes, this is definitely better, thanks!